src/mlpack/methods/ann/ffn.hpp

```diff
@@ -292,6 +318,9 @@ class FFN
 
   //! Locally-stored gradient parameter.
   arma::mat gradient;
+
+  //! Locally-stored dictionary.
+  std::map<std::string, LayerTypes> dictionary;
```

Comment:
This ends up storing two copies of the network.  If anything you should be using a reference to the layers here, not the direct layers themselves.  I suspect also that the code will not work if you build the network, then train it, then try to access a layer by name, since the extraneous copy does not get trained.